consensus/src/main/java/org/apache/iotdb/consensus/multileader/MultiLeaderServerImpl.java

```diff
@@ -108,49 +115,59 @@ public void stop() {
    * records the index of the log and writes locally, and then asynchronous replication is performed
    */
   public TSStatus write(IConsensusRequest request) {
-    synchronized (stateMachine) {
-      if (needToThrottleDown()) {
-        logger.info(
-            "[Throttle Down] index:{}, safeIndex:{}",
-            getIndex(),
-            getCurrentSafelyDeletedSearchIndex());
-        try {
-          stateMachine.wait(config.getReplication().getThrottleTimeOutMs());
-        } catch (InterruptedException e) {
-          logger.error("Failed to throttle down because ", e);
-          Thread.currentThread().interrupt();
+    stateMachineLock.lock();
+    try {
+      synchronized (stateMachine) {
```

Comment:
We can remove this `synchronized` here because we already lock the `stateMachineLock`. On the other hand, when current thread trigger the `await` of condition, it still hold the `stateMachine`'s object lock, which may lead unexpected behaviors.